### DISCUSSION TITLE
Incorrect symlinks for docker (cask) on M1?

### DISCUSSION THREAD

**Discussion by dhduvall** [Homebrew/discussions#2653] [Casks]
I ran `brew install homebrew/cask/docker` last night, pulling what looks like the Homebrew/homebrew-cask@4b5b01d175 version of the cask. It installed mostly as expected, except that, despite being on an M1 Mac, all of the docker commands ended up symlinked into `/usr/local/bin` instead of `/opt/homebrew/bin`. I can't figure out what caused these symlinks to happen in the first place, never mind why they went to the wrong place. The completion scripts that are explicitly mentioned in the cask source (and the only files explicitly listed in `brew ls --cask -v docker`) did go to the right place.

**UPDATE**: I tried `brew uninstall --cask docker`, which removed *most* of the links in `/usr/local/bin` (`com.docker.cli`, `docker-compose-v1`, and `hub-tool` were left behind). I removed those manually, and re-ran `brew install --cask docker`, which put nothing into `/usr/local/bin`. I also tried installing the `docker` formula on top of this, but it didn't put anything into `/opt/homebrew/bin`, either.

```shellsession
$ brew config
HOMEBREW_VERSION: 3.3.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3f0b412951996a675b8a48037e9a978f0ccd8363
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 908f5b47e930643d5c99cf8654ca2bc2191f0d1f
Core tap last commit: 14 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: /opt/homebrew/bin/vi
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2
Rosetta 2: false
```

**Comment by dhduvall** [comment]
I see; running the desktop app for the first time puts those links into `/usr/local/bin`. It's not homebrew at all. I guess then the question is what the behavior should be. Should Docker.app put links into whatever the homebrew directory is? Should it delegate the linking to homebrew if it's installed?

**Reply by SMillerDev** [reply]
It's linking to `/usr/local/bin` because that's convention. The same is true for homebrew. Except homebrew moved because we had a conflict otherwise.
